test/ttsafe.c

```diff
@@ -112,7 +112,7 @@ main(int argc, char *argv[])
 #endif /* H5_HAVE_PTHREAD_H */
     AddTest("acreate", tts_acreate, cleanup_acreate, "multi-attribute creation", NULL);
     AddTest("attr_vlen", tts_attr_vlen, cleanup_attr_vlen, "multi-file-attribute-vlen read", NULL);
-
+    AddTest("mutex", tts_mutex, cleanup_mutex, "mutex operations", NULL);
```

Comment:
Is it intended that this test should run in multi-thread builds of the library as well? If so you may want to either have this in a separate #ifdef outside of `#ifdef H5_HAVE_THREADSAFE` above, or modify that #ifdef to include multi-thread builds, in which case all the tests would run. It was brought up previously that we should be testing the thread-safe tests in multi-thread builds of the library, but that may be outside the scope of this PR if there are any issues with that.

Reply:
I've changed the mutex tests to run whenever the routines themselves are compiled - both in threadsafe and multi-thread builds. I'll leave the synchronization of threadsafe and multithread tests to the test rework PR